### PR TITLE
Remove the duplicate of the auth query parameters

### PR DESCRIPTION
**Pull request by heidijinxujia**
Fix for issue https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/688

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java [hunk 1]
```diff
@@ -153,4 +154,25 @@ public static int compareSemanticVersion(final String thisVersion, final String
...
+     * @param str
+     * @param subString
+     * @return int
+     */
+    public static int countMatches(@NonNull String str, @Nullable String subString) {
+        int count = 0;
+        if (StringUtil.isEmpty(str) || StringUtil.isEmpty(subString)) {
+            return count;
+        }
+        for(int i = 0; i < (str.length() - subString.length()); i++) {
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java]: There is an off by one error here. Need `less than or equal to comparison <=` to get accurate count

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java [hunk 2]
```diff
@@ -139,32 +140,43 @@ public String getPrompt() {
...
 
     @Override
-    public Uri getAuthorizationRequestAsHttpRequest() throws UnsupportedEncodingException {
-        Uri.Builder uriBuilder = Uri.parse(getAuthorizationEndpoint()).buildUpon();
-        for (Map.Entry<String, Object> entry : ObjectMapper.serializeObjectHashMap(this).entrySet()) {
-            uriBuilder.appendQueryParameter(entry.getKey(), entry.getValue().toString());
-        }
+    public Uri getAuthorizationRequestAsHttpRequest() {
+        Map<String, Object> qpMap = new HashMap<>();
+        qpMap.putAll(ObjectMapper.serializeObjectHashMap(this));
```
Comment (Om83) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java]: I am assuming that this is the behavior for iOS, where if the qmap already has the item it is prioritized over the similar  query parameter item.

File: common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java [hunk 3]
```diff
@@ -153,4 +154,25 @@ public static int compareSemanticVersion(final String thisVersion, final String
...
+            if(String.copyValueOf(str.toCharArray(), i, subString.length()).equalsIgnoreCase(subString)) {
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java]: Why are we converting string to char array and then using String.copyValueOf? The string `substring` method can achieve the same result with less work I believe. Is there a reason you chose to do it this way?

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java (hunk 2 above)
Comment (heidijinxujia) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java]: No, actually, iOS does not do any thing on the MSAL calls. iOS only whitelisted all the extra query parameters when doing the broker calls, including authorization request and token request.

**Comment by rpdome** [comment]
Once you're done, can you bump the common version to -RC12? Thanks!

**Pull request merged**